SpiceSharp/Simulations/Exports/RealPropertyExport.cs

```diff
@@ -55,42 +57,54 @@ public RealPropertyExport(IEventfulSimulation simulation, string[] entityPath, s
         /// <param name="e">The <see cref="System.EventArgs" /> instance containing the event data.</param>
         protected override void Initialize(object sender, EventArgs e)
         {
-            e.ThrowIfNull(nameof(e));
-            if (EntityPath.Count == 1)//then path is entity name
-            {
-                var eb = Simulation.EntityBehaviors[EntityPath[0]];
-                if (eb != null)
-                    Extractor = eb.CreatePropertyGetter<double>(PropertyName);
-            }
-            else//using subcircuit array of strings
+            // Collect all behaviors in the most top-level entity
+            var behaviors = new HashSet<IBehavior>();
+            foreach (var behavior in Simulation.EntityBehaviors[EntityPath[0]])
+                behaviors.Add(behavior);
+
+            // For every subsequent name in the path, we need to collect the next set of behaviors
+            for (int i = 1; i < EntityPath.Count; i++)
             {
-                string curComponentName = EntityPath[0];
-                System.Collections.Generic.IEnumerator<Behaviors.IBehavior> entityEnum = Simulation.EntityBehaviors[curComponentName].GetEnumerator();
-                for (int i = 1; i < EntityPath.Count; i++)
-                {
-                    string nextComponentName = EntityPath[i];
+                string nextComponentName = EntityPath[i];
+                // Keep track of the behaviors one level deeper
+                var subBehaviors = new HashSet<IBehavior>();
 
-                    entityEnum.MoveNext();
-                    Components.Subcircuits.EntitiesBehavior behavior = (SpiceSharp.Components.Subcircuits.EntitiesBehavior)entityEnum.Current;
-                    Behaviors.IBehaviorContainerCollection localBehavior = behavior.LocalBehaviors;
-                    foreach (Behaviors.IBehaviorContainer bc in localBehavior)
+                // Go through all the behaviors of the previously found level, and collect the new set of behaviors
+                foreach (var behavior in behaviors)
+                {
+                    if (behavior is SpiceSharp.Components.Subcircuits.EntitiesBehavior subcktBehavior)
                     {
-                        if (bc.Name == nextComponentName)
+                        // Add all the behaviors in this one to the new level of found behaviors
+                        foreach (var behaviorContainer in subcktBehavior.LocalBehaviors)
                         {
-                            if (i == EntityPath.Count - 1)//if last name
-                            {
-                                Extractor = bc.CreatePropertyGetter<double>(PropertyName);
-                                return;
-                            }
-                            else//move to the next subcircuit
+                            if (behaviorContainer.Name == nextComponentName)//only add behaviors that are part of path 
```

Comment:
There is no need to go through all `LocalBehaviors` to find the behaviorContainer with the right name. Simply use `behaviorContainer = subcktBehavior.LocalBehaviors[nextComponentName]` or `subcktBehavior.LocalBehaviors.TryGetBehaviors(nextComponentName, out behaviorContainer)` to get the behavior container associated to the component name directly. This is more efficient as well for searching by name.

Reply:
Something like
```csharp
if (subcktBehavior.LocalBehaviors.TryGetBehaviors(nextComponentName, out behaviorContainer)
{
    foreach (var subBehavior in behaviorContainer)
    {
        // ...
    }
}
```

Reply:
If this is fixed, I think it's going to be :+1: